idem_provider_azurerm/exec/azurerm/compute/virtual_machine.py

```diff
@@ -292,6 +299,27 @@
                 { 'image_reference': dict(zip(image_keys, image.split('|'))) }
             )
 
+    if disk_enc_secret_url:
```

Comment:
this logic has the ability to leave the VM in a weird state. we need to check all 4 variables and only proceed if they're all there. Fail if someone provides 1-3 of the 4 or the resource IDs aren't right.